### PR TITLE
IamDiscovery

### PR DESCRIPTION
**Pull request by PrzemyslawTusinski**
ServiceAccount

```
 {
  "documentId" : "D9b4nlUCNkGYTrMJA7flTA",
  "assetId" : "projects/oss-discovery-test/serviceAccounts/281989297013-compute@developer.gserviceaccount.com",
  "resourceName" : null,
  "resourceId" : null,
  "resourceType" : "GCP::Iam::ServiceAccount",
  "region" : null,
  "accountId" : null,
  "projectId" : "oss-discovery-test",
  "createdIso" : null,
  "updatedIso" : "2021-06-29T14:01:31.700868Z",
  "discoverySessionId" : null,
  "maxSizeInBytes" : null,
  "sizeInBytes" : null,
  "configuration" : {
    "displayName" : "Compute Engine default service account",
    "email" : "281989297013-compute@developer.gserviceaccount.com",
    "etag" : "MDEwMjE5MjA=",
    "name" : "projects/oss-discovery-test/serviceAccounts/281989297013-compute@developer.gserviceaccount.com",
    "oauth2ClientId" : "104780228386383191044",
    "projectId" : "oss-discovery-test",
    "uniqueId" : "104780228386383191044"
  },
  "supplementaryConfiguration" : {
    "iamPolicy" : {
      "etag" : "ACAB"
    },
    "keys" : {
      "keys" : [ {
        "keyAlgorithm" : "KEY_ALG_RSA_2048",
        "keyOrigin" : "GOOGLE_PROVIDED",
        "keyType" : "SYSTEM_MANAGED",
        "name" : "projects/oss-discovery-test/serviceAccounts/281989297013-compute@developer.gserviceaccount.com/keys/cf8bc1b504b690ee8ad8e51678d114c11d51f59e",
        "validAfterTime" : "2021-06-22T12:38:11Z",
        "validBeforeTime" : "2021-07-08T12:38:11Z"
      } ]
    }
  },
  "tags" : { },
  "discoveryMeta" : { }
}
```

Role
```
{
  "documentId" : "ojl4aBvVPFagf-R1y94OBw",
  "assetId" : "roles/apigee.apiAdmin",
  "resourceName" : null,
  "resourceId" : null,
  "resourceType" : "GCP::Iam::Role",
  "region" : null,
  "accountId" : null,
  "projectId" : "oss-discovery-test",
  "createdIso" : null,
  "updatedIso" : "2021-06-29T14:01:33.006657Z",
  "discoverySessionId" : null,
  "maxSizeInBytes" : null,
  "sizeInBytes" : null,
  "configuration" : {
    "description" : "Full read/write access to all apigee API resources",
    "etag" : "AA==",
    "name" : "roles/apigee.apiAdmin",
    "stage" : "GA",
    "title" : "Apigee API Admin"
  },
  "supplementaryConfiguration" : { },
  "tags" : { },
  "discoveryMeta" : { }
}, {
  "documentId" : "sWw9Sw0kOkuVOJUC6cZjVA",
  "assetId" : "roles/apigee.apiReader",
  "resourceName" : null,
  "resourceId" : null,
  "resourceType" : "GCP::Iam::Role",
  "region" : null,
  "accountId" : null,
  "projectId" : "oss-discovery-test",
  "createdIso" : null,
  "updatedIso" : "2021-06-29T14:01:33.006837Z",
  "discoverySessionId" : null,
  "maxSizeInBytes" : null,
  "sizeInBytes" : null,
  "configuration" : {
    "description" : "Reader of apigee resources",
    "etag" : "AA==",
    "name" : "roles/apigee.apiReader",
    "stage" : "GA",
    "title" : "Apigee API Reader"
  },
  "supplementaryConfiguration" : { },
  "tags" : { },
  "discoveryMeta" : { }
} 
```

### CODE REVIEW

**Comment by PrzemyslawTusinski** [comment]
@kickroot  @belosh59  @stefansjfw 
It looks like using REST api instead of java api allows to discover more information and returns type that is easier serializable, resulting in clean configuration json

File: pom.xml [hunk 1]
```diff
@@ -236,6 +236,11 @@
       <groupId>com.google.cloud</groupId>
       <artifactId>google-cloud-dataproc</artifactId>
     </dependency>
+    <dependency>
+      <groupId>com.google.apis</groupId>
+      <artifactId>google-api-services-iam</artifactId>
+      <version>v1-rev20210519-1.31.5</version>
```
Comment (stefansjfw) [inline pom.xml:247]: is stating particular version needed?

Comment (PrzemyslawTusinski) [reply pom.xml:247]: Yes

Comment (PrzemyslawTusinski) [reply pom.xml:247]: It won't compile w/o it

**Review by stefansjfw — APPROVED** [review]

**Pull request merged**